File: R/Source/Histogram.r
Histogram.hist <- function(s) {
  count <- NULL
  for (letter in letters[1:26]) {
    count[letter] <- 0
  }
  for (letter in strsplit(s, "")[[1]]) {
    count[letter] <- count[letter] + 1
  }
  res <- ""
  for (letter in c("u", "w", "x", "z")) {
    if (count[letter] > 0) {
      str <- sprintf("%s  %-6d", letter, count[letter])
      stars <- paste(replicate(count[letter], "*"), collapse = "")
      res <- paste0(res, str, stars, "\n")
    }
  }
  return(substr(res, 0, nchar(res) - 1))
}

File: Php/Source/Histogram.php
<?php

class Histogram
{
	public static function hist(string $s): string
	{
		$count = [];
		foreach (str_split($s) as $letter)
		{
			$count[$letter] = isset($count[$letter]) ? $count[$letter] + 1 : 1;
		}
		$res = [];
		foreach (['u', 'w', 'x', 'z'] as $letter)
		{
			if (isset($count[$letter]))
			{
				$value = $count[$letter];
				$res[] = sprintf('%s  %-5d %s', $letter, $value, str_repeat('*', $value));
			}
		}

		return join("\n", $res);
	}
}
